### PR TITLE
Added ocaml-core

### PR DESCRIPTION
**Pull request by omecamtiv**

### CODE REVIEW

**Comment by thunder-coding** [comment]
Why did you close it?

**Comment by omecamtiv** [comment]
I am very new to github workflow. Not sure how to run tests in github cli. I thought that creating a pull request i can test the compilation of the package.

**Comment by thunder-coding** [comment]
> I am very new to github workflow. Not sure how to run tests in github cli. I thought that creating a pull request i can test the compilation of the package.

As you are a first time contributor, you need approval for workflow runs, basically to prevent abuse of GitHub Actions. So you can re-open and wait until someone from the team approves it

**Comment by omecamtiv** [comment]
> > I am very new to github workflow. Not sure how to run tests in github cli. I thought that creating a pull request i can test the compilation of the package.
> 
> As you are a first time contributor, you need approval for workflow runs, basically to prevent abuse of GitHub Actions. So you can re-open and wait until someone from the team approves it

Ok then I'm re-opening this pull request.

File: packages/ocaml-core/build.sh
Comment (Grimler91) [inline packages/ocaml-core/build.sh:2]: Please make description a bit shorter (max 100 characters)

File: packages/ocaml-core/build.sh
Comment (Grimler91) [inline packages/ocaml-core/build.sh:6]: Should be no need to specify the license file in this case

File: packages/ocaml-core/build.sh
Comment (Grimler91) [inline packages/ocaml-core/build.sh:10]: Have you tried if it is possible to build out of source, i.e. without this line? If not, please try it

File: packages/ocaml-core/build.sh
Comment (Grimler91) [inline packages/ocaml-core/build.sh:11]: This (or rather `--prefix $TERMUX_PREFIX`) is already part of the [default configure args](https://github.com/termux/termux-packages/blob/4f3f1ee43f30c4975dd9893a2067af0fd0db8156/scripts/build/configure/termux_step_configure_autotools.sh#L103), no need to specify it here

**Review by Grimler91 — CHANGES_REQUESTED** [review]
See comments. Also looks like you have accidentally added the source archive (4.12.0.tar.gz) to the commit, please remove it

File: packages/ocaml-core/build.sh
Comment (thunder-coding) [reply packages/ocaml-core/build.sh:6]: https://github.com/termux/termux-packages/blob/4f3f1ee43f30c4975dd9893a2067af0fd0db8156/scripts/build/termux_step_install_license.sh#L28-35

It is needed for LGPL

Comment (Grimler91) [reply packages/ocaml-core/build.sh:6]: We have generic license files for LGPL in termux-licenses: https://github.com/termux/termux-packages/tree/master/packages/termux-licenses/LICENSES. But only for LGPL-{2.0,2.1,3.0}, "LGPL" is not recognised. LICENSE should be LGPL-2.1, based on https://github.com/ocaml/ocaml/blob/trunk/LICENSE

Comment (Grimler91) [reply packages/ocaml-core/build.sh:6]: They do have a paragraph about "special exception" in their license file though, so including it as was done from the beginning is probably better. Please change to:
```
TERMUX_PKG_LICENSE=LGPL-2.1
TERMUX_PKG_LICENSE_FILE=LICENSE
```

**Comment by Grimler91** [comment]
```
make[2]: Entering directory '/home/builder/.termux-build/ocaml-core/src/stdlib'
../boot/ocamlrun ../boot/ocamlc -use-prims ../runtime/primitives -strict-sequence -absname -w +a-4-9-41-42-44-45-48 -g -warn-error A -bin-annot -nostdlib -principal -safe-string -strict-formats  -nopervasives -c camlinternalFormatBasics.mli
../boot/ocamlrun: 3: Syntax error: Unterminated quoted string
make[2]: *** [Makefile:201: camlinternalFormatBasics.cmi] Error 2
```
Either there is some error in the source, or it is trying to run a cross-compiled binary, and in that case it would need a patch and/or a hostbuild step where the tool is built for the host instead

**Comment by omecamtiv** [comment]
Ok I'll dig into this if possible. I'm a newbie in this. Help is always welcomed from your side.

**Comment by thunder-coding** [comment]
From https://github.com/ocaml/ocaml/issues/6861, I am atleast sure that the repository maintainers are least bothered about cross compiling their package

**Comment by xeffyr** [comment]
You may want to look the old ocaml pull request: https://github.com/termux/termux-packages/pull/1757/files

**Pull request closed without merging**
